examples/jsm/utils/TextureUtils.js

```diff
@@ -62,11 +62,33 @@ export function decompress( texture, maxTextureSize = Infinity, renderer = null
 
 	}
 
-	renderer.setSize( Math.min( texture.image.width, maxTextureSize ), Math.min( texture.image.height, maxTextureSize ) );
+	const width = Math.min( texture.image.width, maxTextureSize );
+	const height = Math.min( texture.image.height, maxTextureSize );
+
+	renderer.setSize( width, height );
 	renderer.clear();
+
+	const renderTarget = new WebGLRenderTarget( width, height, {
+		stencilBuffer: true,
+		colorSpace: IS_SRGB ? SRGBColorSpace : LinearSRGBColorSpace,
+	} );
+	renderer.setRenderTarget( renderTarget );
 	renderer.render( _scene, _camera );
+	renderer.setRenderTarget( null );
+
+	const pixelBuffer = new Uint8Array( width * height * 4 );
+	renderer.readRenderTargetPixels(
+		renderTarget,
+		0, 0,
+		width, height,
+		pixelBuffer
+	);
+	const imageData = new ImageData( new Uint8ClampedArray( pixelBuffer ), width, height );
+
+	renderTarget.texture.image = imageData;
+	renderTarget.dispose();
 
-	const readableTexture = new Texture( renderer.domElement );
+	const readableTexture = renderTarget.texture;
```

Comment:
BTW: I like the general idea of the PR to use a render target instead of using the renderer's canvas. This is the more robust approach especially if someone injects a renderer when calling `decompress()`.

Reply:
Why create a new texture? WebGLRenderTarget is not reused so I'm gussing it already does new Texture internally?

Reply:
> Why create a new texture?

Textures of render targets have a flag that indicate they belong to render targets. When using these textures for rendering, they are treated differently. Instead of setting the flag to a different value, it's more clear to just create a new instance.